parquet/src/main/scala/magnolify/parquet/ParquetField.scala

```diff
@@ -309,14 +329,17 @@ object ParquetField {
     pa: ParquetArray
   ): ParquetField[C[T]] = {
     new ParquetField[C[T]] {
-      override val hasAvroArray: Boolean = pa match {
+      // Legacy compat with Magnolify <= 0.7; future versions will remove AvroCompat in favor of
+      // Configuration-based approach
+      @nowarn("cat=deprecation")
+      val groupAvroArrays: Boolean = pa match {
         case ParquetArray.default               => false
         case ParquetArray.AvroCompat.avroCompat => true
       }
```

Comment:
This was probably not ideal. I think we should be able to re-use any `ParquetField` and decide on the behavior when instantiating the parquet type. Here we now mix behaviors with the config.